### PR TITLE
Types refactoring

### PR DESCRIPTION
**Pull request by melniiv**
## Description
- Adding support for child name birth year fields

## Context

[KK-1023](https://helsinkisolutionoffice.atlassian.net/browse/KK-1023)

## Screenshots

[KK-1023]: https://helsinkisolutionoffice.atlassian.net/browse/KK-1023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/child/form/ChildForm.tsx [hunk 1]
```diff
@@ -29,16 +26,7 @@ const schema = yup.object().shape({
...
-  day: number | string;
-  month: number | string;
-  year: number | string;
-}
-
-interface ChildFormValues extends Omit<Child, 'birthdate'> {
-  birthdate: Birthdate;
-  childBirthdate?: string;
-}
+type ChildFormValues = Child;
```
Comment (nikomakela) [inline src/domain/child/form/ChildForm.tsx:29]: is this reinit really needed?

File: src/domain/child/modal/ChildFormModal.tsx [hunk 2]
```diff
@@ -6,14 +6,7 @@ import { Child } from '../types/ChildInputTypes';
...
 
-export interface ChildFormModalValues extends Omit<Child, 'birthdate'> {
-  birthdate: {
-    day: number | string;
-    month: number | string;
-    year: number | string;
-  };
-  childBirthdate?: string;
-}
+export type ChildFormModalValues = Child;
```
Comment (nikomakela) [inline src/domain/child/modal/ChildFormModal.tsx:9]: Is this reinit really needed?

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

File: src/domain/profile/children/child/ProfileChild.tsx [hunk 3]
```diff
@@ -37,8 +37,9 @@ const ProfileChild: React.FunctionComponent<ProfileChildProps> = ({
 
   const upcomingEventsAndEventGroups =
     child.upcomingEventsAndEventGroups?.edges[0]?.node?.name;
-  const isNamed = Boolean(child.firstName);
-  const childName = `${child.firstName} ${child.lastName}`;
+  const name = child.name.split(' ');
+  const isNamed = Boolean(name[0]);
```
Comment (nikomakela) [inline src/domain/profile/children/child/ProfileChild.tsx:41]: wouldn't `!!name` do the same? Using index is a bit strange imho.

File: src/domain/profile/children/child/ProfileChildDetail.tsx [hunk 4]
```diff
@@ -91,9 +91,7 @@ const ProfileChildDetail = () => {
         <>
           <ListPageLayout.Header
             title={
-              child.firstName
-                ? `${child.firstName} ${child.lastName}`
-                : t('profile.child.default.name.text')
+              child.name ? child.name : t('profile.child.default.name.text')
```
Comment (nikomakela) [inline src/domain/profile/children/child/ProfileChildDetail.tsx:94]: `child?.name || t('profile.child.default.name.text')`

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/domain/registration/form/RegistrationForm.tsx [hunk 5]
```diff
@@ -138,9 +138,20 @@ const RegistrationForm = () => {
...
+                // still birthdate, firstName and lastName returned from values for some reason
+                const childValues = child as any;
+
+                return {
+                  name: `${childValues.firstName} ${childValues.lastName}`,
+                  languagesSpokenAtHome: childValues.languagesSpokenAtHome,
+                  birthyear: childValues.birthyear,
+                  postalCode: childValues.postalCode,
+                  relationship: childValues.relationship,
+                };
```
Comment (nikomakela) [inline src/domain/registration/form/RegistrationForm.tsx:153]: I can check tomorrow

File: src/domain/registration/notEligible/NotEligibleUtils.ts [hunk 6]
```diff
@@ -1,24 +1,20 @@
...
+  validator: (value: any) => boolean;
+  item: any;
+};
+
+/** isBirthyearEligible()
+ * Check if child's birthyear is eligible for participation.
  *
  * Only children born in 2020 or later are eligible for this service.
- * @param {string} birthdate in YYYY-MM-DD format.
+ * @param {number} birthyear in YYYY-MM-DD format.
```
Comment (nikomakela) [inline src/domain/registration/notEligible/NotEligibleUtils.ts:13]: yyyy-mm-dd -> yyyy

File: src/domain/registration/state/RegistrationReducers.tsx [hunk 7]
```diff
@@ -12,10 +12,9 @@ export const defaultRegistrationData: RegistrationData = {
   formValues: {
     children: [
       {
-        birthdate: '',
-        firstName: '',
+        birthyear: 2020,
```
Comment (nikomakela) [inline src/domain/registration/state/RegistrationReducers.tsx:15]: I think current year would be better

**Review by nikomakela — COMMENTED** [review]
Looking really good! Some small notes given. I can help with the form context tomorrow

File: codegen.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import { CodegenConfig } from '@graphql-codegen/cli';
 
 const config: CodegenConfig = {
-  schema: process.env.REACT_APP_API_URI,
+  schema: process.env.VITE_API_URI,
```
Comment (karisal-anders) [inline codegen.ts:4]: This is in `master` branch already so just rebasing will get rid of this. But it won't probably hurt either if you merge this as is as then it'll just throw your change away as already applied I think.

File: src/common/time/TimeConstants.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 // Only support kids which was born starting in 2020 or later
-export const SUPPORTED_START_BIRTHDATE = '2020-01-01';
+export const SUPPORTED_START_BIRTHDATE = 2020;
```
Comment (karisal-anders) [inline src/common/time/TimeConstants.ts:2]: BIRTHDATE → BIRTHYEAR

File: src/common/translation/i18n/fi.json [hunk 10]
```diff
@@ -559,7 +559,7 @@
       "child": {
         "birthdate": {
```
Comment (karisal-anders) [inline src/common/translation/i18n/fi.json:560]: birthdate -> birthyear

File: src/common/translation/i18n/fi.json [hunk 11]
```diff
@@ -559,7 +559,7 @@
...
           "input": {
-            "label": "Lapsen syntymäaika (pp.kk.vvvv)",
+            "label": "Lapsen syntymäaika (vvvv)",
```
Comment (karisal-anders) [inline src/common/translation/i18n/fi.json:562]:
```suggestion
            "label": "Lapsen syntymävuosi",
```

File: src/common/translation/i18n/sv.json [hunk 12]
```diff
@@ -559,7 +559,7 @@
       "child": {
         "birthdate": {
           "input": {
-            "label": "Barnets födelsedatum (dd.mm.åååå)",
+            "label": "Barnets födelsedatum (åååå)",
```
Comment (karisal-anders) [inline src/common/translation/i18n/sv.json:562]:
```suggestion
            "label": "Barnets födelseår",
```

File: src/common/translation/i18n/en.json [hunk 13]
```diff
@@ -559,8 +559,8 @@
       "child": {
         "birthdate": {
```
Comment (karisal-anders) [inline src/common/translation/i18n/en.json:560]: birthdate -> birthyear

File: src/common/translation/i18n/sv.json [hunk 14]
```diff
@@ -559,7 +559,7 @@
       "child": {
         "birthdate": {
```
Comment (karisal-anders) [inline src/common/translation/i18n/sv.json:560]: birthdate -> birthyear

**Review by karisal-anders — COMMENTED** [review]
Some thoughts:
- **All** **birthdate** stuff would be good to be **replaced** with **birthyear**, e.g. childBirthdate -> childBirthyear
- It would be good to be **consistent** with the **writing** of **birthyear**, so I'd suggest to pick one (e.g. birthyear as it seems to be most used in this repo at the moment) and go with that, so birth_year -> birthyear, birthYear -> birthyear.

File: src/common/components/form/validationUtils.tsx [hunk 15]
```diff
@@ -46,8 +49,30 @@ const validateDate = (value: string | number) => {
   }
 };
 
+/** validateyear()
```
Comment (karisal-anders) [inline src/common/components/form/validationUtils.tsx:52]: `validateyear()` in the comment is unnecessary and out-of-sync with the function name, how about just removing this?

File: src/common/components/form/validationUtils.tsx [hunk 16]
```diff
@@ -46,8 +49,30 @@ const validateDate = (value: string | number) => {
...
+ * Validate input year. Dates in the future are not valid.
+ * @param {number} value Input value.
+ * @returns {string} translation key
```
Comment (karisal-anders) [inline src/common/components/form/validationUtils.tsx:55]:
```suggestion
 * Validate input year. Dates in the future are not valid.
 * @param {number} value Input value.
 * @returns {string} translation key if validation error, otherwise empty string
```

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/child/modal/ChildFormModal.tsx (hunk 2 above)
Comment (melniiv) [reply src/domain/child/modal/ChildFormModal.tsx]: fixed

File: src/domain/profile/children/child/ProfileChild.tsx (hunk 3 above)
Comment (melniiv) [reply src/domain/profile/children/child/ProfileChild.tsx]: fixed

File: src/domain/profile/children/child/ProfileChildDetail.tsx (hunk 4 above)
Comment (melniiv) [reply src/domain/profile/children/child/ProfileChildDetail.tsx:94]: fixed

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/registration/notEligible/NotEligibleUtils.ts (hunk 6 above)
Comment (melniiv) [reply src/domain/registration/notEligible/NotEligibleUtils.ts]: fixed

File: codegen.ts (hunk 8 above)
Comment (melniiv) [reply codegen.ts:4]: ok, i thought i had up to date master

File: src/common/time/TimeConstants.ts (hunk 9 above)
Comment (melniiv) [reply src/common/time/TimeConstants.ts]: fixed

File: src/common/translation/i18n/fi.json (hunk 10 above)
Comment (melniiv) [reply src/common/translation/i18n/fi.json]: fixed. but still i would want to go through translations

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

File: src/common/translation/i18n/sv.json (hunk 12 above)
Comment (melniiv) [reply src/common/translation/i18n/sv.json]: fixed

File: src/common/translation/i18n/en.json (hunk 13 above)
Comment (melniiv) [reply src/common/translation/i18n/en.json]: fixed

File: src/common/translation/i18n/sv.json (hunk 14 above)
Comment (melniiv) [reply src/common/translation/i18n/sv.json]: fixed

File: src/common/components/form/validationUtils.tsx (hunk 15 above)
Comment (melniiv) [reply src/common/components/form/validationUtils.tsx]: ?

File: src/common/components/form/validationUtils.tsx (hunk 16 above)
Comment (melniiv) [reply src/common/components/form/validationUtils.tsx]: fixed

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/child/form/ChildForm.tsx (hunk 1 above)
Comment (melniiv) [reply src/domain/child/form/ChildForm.tsx]: fixed

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/common/components/form/validationUtils.tsx (hunk 15 above)
Comment (karisal-anders) [reply src/common/components/form/validationUtils.tsx]: It's a minor thing, just that the function is `validateYear(value: number)` and the comment said simply `validateyear()` so the name was off (i.e. validateYear is not validateyear, the case is off). What do you see as the value the duplication of the function name in the comment brings? I see it as unnecessary duplication myself because IDEs should show the real function name anyway and actually bringing negative value as it can be out-of-sync with the real function name and thus misleading.

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

File: src/common/translation/i18n/sv.json [hunk 17]
```diff
@@ -480,15 +480,9 @@
       }
     },
     "preliminaryForm": {
-      "childBirthdate": {
+      "childBirthYear": {
```
[the remark this answers is not in the document]
Comment (melniiv) [reply src/common/translation/i18n/sv.json]: fixed

File: src/common/translation/i18n/fi.json [hunk 18]
```diff
@@ -480,17 +480,11 @@
       }
     },
     "preliminaryForm": {
-      "childBirthdate": {
+      "childBirthYear": {
```
[the remark this answers is not in the document]
Comment (melniiv) [reply src/common/translation/i18n/fi.json]: fixed

File: src/common/translation/i18n/en.json [hunk 19]
```diff
@@ -480,15 +480,9 @@
       }
     },
     "preliminaryForm": {
-      "childBirthdate": {
+      "childBirthYear": {
```
[the remark this answers is not in the document]
Comment (melniiv) [reply src/common/translation/i18n/en.json]: fixed

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/common/translation/i18n/sv.json [hunk 20]
```diff
@@ -175,7 +175,7 @@
         "availableEventGroup": {
           "readMoreButton": "Bläddra i händelser"
         },
-        "birthdate": "Barnets födelsedatum",
+        "birthyear": "Barnets födelsedatum FIXME",
```
Comment (nikomakela) [inline src/common/translation/i18n/sv.json:178]:
```suggestion
        "birthyear": "Barnets födelseår",
```

File: src/common/translation/i18n/sv.json [hunk 21]
```diff
@@ -480,15 +480,9 @@
...
+      "childBirthyear": {
         "input": {
-          "day": {
-            "placeholder": "datum"
-          },
-          "label": "Barnets födelsedatum (dd.mm.åååå)",
-          "month": {
-            "placeholder": "mån"
-          },
+          "label": "Barnets födelsedatum (dd.mm.åååå) FIXME",
```
Comment (nikomakela) [inline src/common/translation/i18n/sv.json:485]:
```suggestion
          "label": "Barnets födelseår",
```

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by karisal-anders** [comment]
@melniiv Please could you rebase on most recent master branch so it's easier to see the changes from the current master in the UI? Thank you!

```bash
git checkout types-refactoring
git fetch --all
git rebase origin/master
git push origin --set-upstream types-refactoring --force
```

I tried rebasing myself locally and it went through without merge conflicts.

File: src/common/translation/i18n/fi.json [hunk 22]
```diff
@@ -480,17 +480,11 @@
...
+      "childBirthyear": {
         "input": {
-          "day": {
-            "placeholder": "pv"
-          },
-          "label": "Lapsen syntymäaika (pp.kk.vvvv)",
-          "month": {
-            "placeholder": "kk"
-          },
+          "label": "Lapsen syntymäaika (vvvv)",
```
Comment (karisal-anders) [inline src/common/translation/i18n/fi.json:485]:
```suggestion
          "label": "Lapsen syntymävuosi",
```
i.e. "Child's birthyear" in Finnish

File: src/common/translation/i18n/fi.json [hunk 23]
```diff
@@ -557,16 +551,16 @@
         }
       },
       "child": {
-        "birthdate": {
+        "birthyear": {
           "input": {
-            "label": "Lapsen syntymäaika (pp.kk.vvvv)",
-            "placeholder": "Lisää lapsen syntymäaika"
+            "label": "Lapsen syntymäaika (vvvv)",
```
Comment (karisal-anders) [inline src/common/translation/i18n/fi.json:556]:
```suggestion
            "label": "Lapsen syntymävuosi",
```
i.e. "Child's birthyear" in Finnish

File: src/common/translation/i18n/sv.json [hunk 24]
```diff
@@ -557,16 +551,16 @@
         }
       },
       "child": {
-        "birthdate": {
+        "birthyear": {
           "input": {
-            "label": "Barnets födelsedatum (dd.mm.åååå)",
-            "placeholder": "Ange barnets födelsedatum"
+            "label": "Barnets födelsedatum (åååå)",
```
Comment (karisal-anders) [inline src/common/translation/i18n/sv.json:556]:
```suggestion
            "label": "Barnets födelseår",
```
i.e. "Child's birthyear" in Swedish

File: src/common/translation/i18n/sv.json [hunk 25]
```diff
@@ -480,15 +480,9 @@
...
-          "day": {
-            "placeholder": "datum"
-          },
-          "label": "Barnets födelsedatum (dd.mm.åååå)",
-          "month": {
-            "placeholder": "mån"
-          },
+          "label": "Barnets födelseår",
           "year": {
             "placeholder": "år"
```
Comment (karisal-anders) [inline src/common/translation/i18n/sv.json:487]:
```suggestion
            "placeholder": "Ange barnets födelseår"
```
i.e. "Enter the child's year of birth" in Swedish, making the placeholder more similar to other places in the UI

File: src/common/translation/i18n/en.json [hunk 26]
```diff
@@ -480,15 +480,9 @@
...
-          "day": {
-            "placeholder": "date"
-          },
-          "label": "Child's date of birth (dd.mm.yyyy)",
-          "month": {
-            "placeholder": "month"
-          },
+          "label": "Child's year of birth",
           "year": {
             "placeholder": "year"
```
Comment (karisal-anders) [inline src/common/translation/i18n/en.json:487]:
```suggestion
            "placeholder": "Enter child's year of birth"
```
to make the placeholder more similar to other placeholders in the UI

File: src/common/translation/i18n/fi.json [hunk 27]
```diff
@@ -480,17 +480,11 @@
...
-            "placeholder": "pv"
-          },
-          "label": "Lapsen syntymäaika (pp.kk.vvvv)",
-          "month": {
-            "placeholder": "kk"
-          },
+          "label": "Lapsen syntymäaika (vvvv)",
           "year": {
-            "placeholder": "vuosi"
+            "placeholder": "Vuosi"
```
Comment (karisal-anders) [inline src/common/translation/i18n/fi.json:487]:
```suggestion
            "placeholder": "Lisää lapsen syntymävuosi"
```
i.e. "Enter child’s last name" in Finnish, to make the placeholder more similar to other placeholders in the UI

File: src/domain/registration/form/RegistrationForm.tsx [hunk 28]
```diff
@@ -138,9 +133,17 @@ const RegistrationForm = () => {
               setFormIsFilling(false);
               dispatch(setFormValues(values));
 
-              const backendSupportChildren = values.children.map((child) =>
-                getSupportedChildData(child)
-              );
+              const backendSupportChildren = values.children.map((child) => {
+                // todo: test if its working
+                const childValues = child as any;
```
Comment (karisal-anders) [inline src/domain/registration/form/RegistrationForm.tsx:138]: Why is this needed? Still in testing, right?

File: src/domain/registration/state/RegistrationReducers.tsx [hunk 29]
```diff
@@ -12,10 +12,9 @@ export const defaultRegistrationData: RegistrationData = {
   formValues: {
     children: [
       {
-        birthdate: '',
-        firstName: '',
+        birthyear: new Date().getFullYear(),
```
Comment (karisal-anders) [inline src/domain/registration/state/RegistrationReducers.tsx:15]: It's debatable whether it's really good to have the current year here as a default value. I'm vote for not having a default value here at all as it is always incorrect.

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Review by nikomakela — APPROVED** [review]
It's now approved! Good job!

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/domain/registration/form/RegistrationForm.tsx [hunk 30]
```diff
@@ -137,10 +140,16 @@ const RegistrationForm = () => {
             onSubmit={(values) => {
               setFormIsFilling(false);
               dispatch(setFormValues(values));
-
-              const backendSupportChildren = values.children.map((child) =>
-                getSupportedChildData(child)
-              );
+              const backendSupportChildren = values.children.map((child) => {
+                const childValues = child as any;
```
Comment (karisal-anders) [inline src/domain/registration/form/RegistrationForm.tsx:144]: Why is this casting as any needed?

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kukkuu-local.api.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

File: src/domain/registration/form/RegistrationForm.tsx (hunk 30 above)
Comment (melniiv) [reply src/domain/registration/form/RegistrationForm.tsx]: fixed

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-ui-pr523.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## TestCafe result is __failed__ for https://kukkuu-local.api.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Review by karisal-anders — APPROVED** [review]

**Comment by terovirtanen** [comment]
**KUKKUU-UI branch is deployed to platta: https://kukkuu-ui-pr523.dev.hel.ninja** :rocket::rocket::rocket:
